tools/rename_apk_aab.sh

```diff
@@ -0,0 +1,111 @@
+#!/bin/bash
+
+# This script aims to rename any `.apk` or `.aab` file to `[wpandroid|jpandroid]-{version}[-Signed].apk/aab`
+# depending on the APK/AAB package name, version name and package signature.
+#
+# The main use of this tool os to easily rename the signed `.apk` and/or original `.aab` downloaded from the PlayStore
+# (and re-signed by Google depending on the context), as when you download those their basename is usually just the
+# versionCode and whose name doesn't distinguish from a binary corresponding to WordPress vs Jetpack.
+#
+# By running this script on the APK or AAB files provided – defaulting to all the APK and AAB files in `~/Downloads` if
+# no parameter provided – the files will be renamed with a basename appropriate to the ones we use when attaching
+# those files to the GitHub release.
+
+### Input Parameters ###
+
+# Usage:
+#   rename_apk_aab.sh [file1.apk] [file2.aab] [...]
+#
+# Without any parameter, applies the rename to every APK/AAB file found in ~/Downloads
+#
+INPUT_FILES=( "$@" )
+if [[ $# -lt 1 ]]; then
+  IFS=$'\n'
+  INPUT_FILES=( $(find ~/Downloads -name *.apk -o -name *.aab) )
+  unset IFS
+fi
+
+
+##################################
+### Path to Android tools used ###
+##################################
+AAPT2=$(ls -1t $ANDROID_SDK_ROOT/build-tools/*/aapt2 | head -n1)
+APKSIGNER=$(ls -1t $ANDROID_SDK_ROOT/build-tools/*/apksigner | head -n1)
+BUNDLETOOL="/usr/local/bin/bundletool"
+
+######################
+### Helper Methods ###
+######################
+
+### Extract info from a single APK file
+# Sets PKG, VNAME, VCODE and SIGNED_SUFFIX
+info_for_apk() {
+  # Use aapt2 to extract package name, versionCode and versionName
+  INFO_LINE=$($AAPT2 dump badging "$1" | head -n 1)
+  [[ "$INFO_LINE" =~ name=\'([^\']*)\'[[:blank:]]versionCode=\'([^\']*)\'[[:blank:]]versionName=\'([^\']*)\' ]] && PKG=${BASH_REMATCH[1]} && VCODE=${BASH_REMATCH[2]} && VNAME=${BASH_REMATCH[3]}
+  SIGNED_SUFFIX=$($APKSIGNER verify --print-certs "$1" | grep -qE "(CN=Android, OU=Android, O=Google Inc.)|(O=Automattic Inc.)" && echo "-Signed")
```

Comment:
I was surprised to discover that the signed APK – the one downloaded from the Google PlayStore Console, so potentially resigned by PlayStore) was:
 - Signed by a Google certificate for the Jetpack APK
 - Signed by an Automattic certificate for the WordPress APK

Then I remembered that while when we created the app in PlayStore Console for WordPress, we did upload the a8c signing certificate, but when we created the app for Jetpack, we didn't (step was missed during setup), and Google auto-created a certificate for us.
While we've pondered back then (at the beginning of us setting up Jetpack) of revoking the Google cert and providing our own instead, we realized that it would have more harmful consequences than anything (invalidating previous builds and making the app not auto-update for people who already downloaded it) so we kept it that way.
Which ultimately explains why JP is signed with Google but WP with Automattic.